### PR TITLE
Bump cilium-cli to v0.8.4

### PR DESCRIPTION
**Pull request by tklauser**

### CODE REVIEW

**Comment by gandro** [comment]
With v0.8.4 we should be able to remove these `FIXME` lines and the `timeout 5s` part here:
https://github.com/cilium/cilium/blob/394c084e20344ba151c7ecc4b89620094b4024e3/.github/workflows/conformance-multicluster.yaml#L281-L285

I can do it in a separate PR though if it's already too late.

**Comment by tklauser** [comment]
> With v0.8.4 we should be able to remove these `FIXME` lines and the `timeout 5s` part here:
> 
> https://github.com/cilium/cilium/blob/394c084e20344ba151c7ecc4b89620094b4024e3/.github/workflows/conformance-multicluster.yaml#L281-L285
> 
> I can do it in a separate PR though if it's already too late.

Thanks for pointing this out. I'll add a separate commit removing these `FIXME`s as I likely will have to cherry-pick a fix for the multicluster workflow anyway.

**Comment by tklauser** [comment]
Added a commit to address https://github.com/cilium/cilium/pull/16799#issuecomment-874772363 and cherry-picked @nbusseneau's commit from #16787 fixing the multicluster workflow. Hopefully all required actions should pass now.

File: .github/workflows/conformance-multicluster.yaml [hunk 1]
```diff
@@ -278,11 +278,7 @@ jobs:
         if: ${{ failure() }}
         run: |
           cilium status --context ${{ steps.contexts.outputs.context1 }}
-          # FIXME: Use of timeout is a workaround for cilium/cilium-cli#384
-          timeout 5s cilium clustermesh status --context ${{ steps.contexts.outputs.context1 }}
           cilium status --context ${{ steps.contexts.outputs.context2 }}
-          # FIXME: Use of timeout is a workaround for cilium/cilium-cli#384
-          timeout 5s cilium clustermesh status --context ${{ steps.contexts.outputs.context2 }}
```
Comment (gandro) [inline .github/workflows/conformance-multicluster.yaml:298]: We still want to dump the status, but thanks to cilium/cilium-cli#384 it won't block forever anymore, so we can just remove the timeout:

```suggestion
          cilium clustermesh status --context ${{ steps.contexts.outputs.context2 }}
```

Comment (tklauser) [reply .github/workflows/conformance-multicluster.yaml:298]: :facepalm: thanks, changed

**Comment by tklauser** [comment]
Waiting for #16787 (which should now fix the failing multicluster test) to be reviewed and merged and then will rebase this PR.

**Review by michi-covalent — APPROVED** [review]

File: .github/workflows/conformance-multicluster.yaml (hunk 1 above)
Comment (gandro) [reply .github/workflows/conformance-multicluster.yaml:298]: Note, this doesn't seem to be fixed on the branch yet.

**Review by nbusseneau — COMMENTED** [review]
Same remark as Sebastian, we need both `cilium clustermesh status` back in, otherwise LGTM :D

File: .github/workflows/conformance-multicluster.yaml (hunk 1 above)
Comment (tklauser) [reply .github/workflows/conformance-multicluster.yaml:298]: :facepalm: pushed the wrong copy of the branch. Should now be fixed.

**Review by nbusseneau — APPROVED** [review]

File: .github/workflows/conformance-multicluster.yaml [hunk 2]
```diff
@@ -277,7 +277,7 @@ jobs:
         run: |
           cilium hubble port-forward --context ${{ steps.contexts.outputs.context1 }}&
           sleep 10s
-          [[ $(pgrep -f "cilium hubble port-forward|kubectl port-forward.*hubble-relay" | wc -l) == 2 ]]
+          [[ $(pgrep -f "cilium hubble port-forward|kubectl port-forward.*hubble-relay" | wc -l) == 4 ]]
```
Comment (nbusseneau) [inline .github/workflows/conformance-multicluster.yaml]: > `cilium hubble port-forward` is run for both clusters, so we expect to see each process 2 times.

Wait, am I missing something? We don't run `cilium hubble port-forward` for the second cluster, and it doesn't even have Relay 🤔

Comment (tklauser) [reply .github/workflows/conformance-multicluster.yaml]: You're completely right, sorry. I must have mixed something up. I'll drop the commit again. Though it's still a mystery to me why that check is consistently failing.

**Comment by tklauser** [comment]
Blocked on #16831

**Comment by tklauser** [comment]
Rebased to pick up #16787 and #16831, also addressed Sebastian's remark. Tests are all passing now :tada:                                                          
  
* AKS https://github.com/cilium/cilium/actions/runs/1015048476
* EKS https://github.com/cilium/cilium/actions/runs/1015048468
* AWS-CNI https://github.com/cilium/cilium/actions/runs/1015048464
* GKE https://github.com/cilium/cilium/actions/runs/1015048472
* Multicluster https://github.com/cilium/cilium/actions/runs/1015048478
* Kind https://github.com/cilium/cilium/actions/runs/1015048465

Removing the temporary test commit and moving out of draft.

**Review by nbusseneau — APPROVED** [review]
:shipit:

**Review by aanm — APPROVED** [review]

**Review by christarazi — APPROVED** [review]

**Comment by nbusseneau** [comment]
All reviews are in, PR only touches CI 3.0 workflows, and all of them have been manually tested via `pull_request` triggers (cf. [above](https://github.com/cilium/cilium/pull/16799#issuecomment-877172068)). Marking as `ready-to-merge`.

**Pull request merged**